### PR TITLE
Basic blockchain implementation

### PR DESCRIPTION
**Pull request by adrianryt**

### CODE REVIEW

**Review by xsanm — APPROVED** [review]
LGTM 💪

File: src/main/java/logic/App.java [hunk 1]
```diff
@@ -17,5 +20,14 @@ public static void main(String[] args) {
         AbstractTransaction t1 = TransactionFactory.getSellBuySharesTransaction(i, new Date(),2,"Ja",
                 TransactionType.SharesBuySell, "GIT", Priority.HIGH, "Adam", "Tobys", 5);
         System.out.println(t1);
+
+
+        Gson parser = new Gson();
+        String serializedTransaction = parser.toJson(t1);
+        System.out.println(serializedTransaction);
+        System.out.println("DUPA1");
```
Comment (wawel37) [inline src/main/java/logic/App.java:28]: Hmmm

File: src/main/java/logic/App.java [hunk 2]
```diff
@@ -17,5 +20,14 @@ public static void main(String[] args) {
...
+        AbstractTransaction t2 = parser.fromJson(serializedTransaction, SellBuyShares.class);
+        System.out.println("DUPA2");
```
Comment (wawel37) [inline src/main/java/logic/App.java:30]: hmmm2

**Review by wawel37 — APPROVED** [review]
small nit changes

File: src/main/java/logic/App.java (hunk 2 above)
Comment (xsanm) [reply src/main/java/logic/App.java:30]: @adrianryt  that's disgusting 😢

**Pull request merged**